test/e2e/framework/util.go

```diff
@@ -1456,6 +1456,7 @@ func waitForPodTerminatedInNamespace(c *client.Client, podName, reason, namespac
 func waitForPodSuccessInNamespaceTimeout(c *client.Client, podName string, contName string, namespace string, timeout time.Duration) error {
 	return waitForPodCondition(c, namespace, podName, "success or failure", timeout, func(pod *api.Pod) (bool, error) {
 		// Cannot use pod.Status.Phase == api.PodSucceeded/api.PodFailed due to #2632
+		// TODO(random-liu): This was not true from long time ago. We can use api.PodSucceeded now.
```

Comment:
I saw this last time as well. Was thinking about fixing it, but so many tests are using it :\
Are you sure you want to put your name in the TODO?